packages/server/validators/events.js

```diff
@@ -0,0 +1,203 @@
+    .trim()
+    .escape()
+    .isISO8601()
+    .toDate()
+    .withMessage("Start date is not in a valid date format")
+    .custom(async (startDate, { req }) => {
+      //If new end date is provided, check if start date is before the end date
+      if (req.body.endDate) {
+        if (startDate > req.body.endDate) {
+          throw new Error("Start date is after new end date");
+        }
+      }
+    }),
+
+  //End Date checks
+  //Similar checks to start sate but makes sure end date is after the start
+  check("startDate").toDate(),
+  body("endDate", "Invalid end date")
+    .trim()
+    .escape()
+    .isISO8601()
+    .toDate()
+    .withMessage("End date is not in a valid date format")
+    .custom(async (endDate, { req }) => {
+      if (req.body.startDate) {
+        if (endDate < req.body.startDate) {
+          throw new Error("End date is before new start date");
+        }
+      }
+    }),
+];
+
+const updateEventValidator = [
+  //Title checks
+  body("title", "Invalid title")
+    .trim()
+    .escape()
+    .optional()
+    .isLength({ min: 1, max: 255 })
+    .withMessage("Title length must be between 1 to 255 characters"),
+  //Description checks
+  body("description", "Invalid description")
+    .trim()
+    .escape()
+    .optional()
+    .isLength({ max: 2000 })
+    .withMessage("Description max length is 2000 characters"),
+  //Location checks
+  body("location", "Invalid location")
+    .trim()
+    .escape()
+    .optional()
+    .isLength({ max: 255 })
+    .withMessage("Location max length is 255 characters"),
+  //Thumbnail checks
+  body("thumbnail", "Invalid thumbnail")
+    .trim()
+    .escape()
+    .optional()
+    .isLength({ max: 255 })
+    .withMessage("Thumbnail max length is 255 characters")
+    .custom(async (value) => {
+      if (!/\.(jpg|png|JPG|PNG|JPEG|jpeg)$/.test(value)) {
+        throw new Error("Thumbnail is not a valid image file");
+      }
+    }),
+  //Start Date checks
+  //Convert end date to a Date object for easier comparison
+  check("endDate").optional().toDate(),
+  body("startDate", "Invalid dates")
+    .trim()
+    .escape()
+    .optional()
+    .isISO8601()
+    .toDate()
+    .withMessage("Start date is not in a valid date format")
+    .custom(async (startDate, { req }) => {
+      //If new end date is provided, check if start date is before the end date
+      if (req.body.endDate) {
+        if (startDate > req.body.endDate) {
+          throw new Error("Start date is after new end date");
+        }
+      } else {
+        //Checks if new start date is before kept end date
+        const currEvent = await EventController.getEvent(req.params.eventId);
+        if (currEvent.endDate != null && startDate > currEvent.endDate) {
+          throw new Error("Start date is after end date");
+        }
+      }
+    }),
+  //End Date checks
+  //Similar checks to start sate but makes sure end date is after the start
+  check("startDate").optional().toDate(),
```

Comment:
We used the checks to actually convert the dates from strings to date objects. Earlier we were having trouble accessing and converting the data but I think we found an alternate solution that lets us avoid using check() for that. In the case of updating the start date, I found using the if() for conditional validation was not helpful. When the if() evaluates to false it would ignore any validations after it, effectively ending the validation early for that value. Since I needed to run certain checks based on if the end date was present, I found doing a simple if-else statement in the custom validator to be a better solution. For updating the end date, I also followed that solution to keep them consistent with each other.